packages/console/src/contexts/SubscriptionDataProvider/use-new-subscription-data.ts

```diff
@@ -52,11 +51,14 @@ const useNewSubscriptionData: () => NewSubscriptionContext & { isLoading: boolea
 
   useEffect(() => {
     if (subscriptionUsageData?.quota) {
-      updateTenant(tenantId, {
+      updateTenant(currentTenantId, {
         quota: pick(subscriptionUsageData.quota, 'mauLimit', 'tokenLimit'),
       });
     }
-  }, [tenantId, subscriptionUsageData?.quota, updateTenant]);
+    // Since `updateTenant` updates `tenants` in TenantsContext and triggers re-renders (`updateTenant` is re-rendered as well),
```

Comment:
Can we use `useCallback` to resolve this?

Reply:
Tested with `useCallback()` locally, it worked fine.